fs/rpmsgfs/rpmsgfs_client.c

```diff
@@ -740,8 +741,8 @@ int rpmsgfs_client_rename(FAR void *handle, FAR const char *oldpath,
 
   DEBUGASSERT(len <= space);
 
-  strcpy(msg->pathname, oldpath);
-  strcpy(msg->pathname + oldlen, newpath);
+  memcpy(msg->pathname, oldpath, oldlen);
+  memcpy(msg->pathname + oldlen, newpath, newlen);
 
```

Comment:
Are we supposed to leave one NUL byte between oldpath and newpath in the buffer? (Asking because I'm not familiar with this protocol.)

Reply:
Yes, since this format can avoid the rpmsgfs server copy the oldpath temporally to append the NUL terminator.